### PR TITLE
ARROW-1721: [Python] Implement null-mask check in places where it isn't supported in numpy_to_arrow.cc

### PR DESCRIPTION
**Pull request by Licht-T**
This closes [ARROW-1721](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-1721).

### CODE REVIEW

File: cpp/src/arrow/python/numpy_to_arrow.cc [hunk 1]
```diff
@@ -1029,6 +1033,44 @@ Status LoopPySequence(PyObject* sequence, T func) {
   return Status::OK();
 }
 
+template <typename T>
+Status LoopPySequenceWithMasks(
+    PyObject* sequence,
+    const Ndarray1DIndexer<uint8_t>& mask_values,
+    bool have_mask,
+    T func
+) {
```
Comment (wesm) [inline cpp/src/arrow/python/numpy_to_arrow.cc]: Can you run clang-format? (`make format` or `ninja format`). This should also fix the cpplint failure in CI . See https://github.com/apache/arrow/tree/master/cpp#continuous-integration

**Comment by wesm** [comment]
Thank you for doing this!

**Comment by Licht-T** [comment]
@wesm Fixed the whole lint issues!

**Comment by wesm** [comment]
I'm surprised by some of the formatting changes, are you using clang-format-4.0?

**Comment by wesm** [comment]
I ran clang-format 4.0 locally and got this diff https://github.com/wesm/arrow/commit/7547ac8e70b5279e44fe802bdbd241ad9a8f0d4a

**Comment by Licht-T** [comment]
@wesm Sorry! I'm using clang-format-5!
I'll fix!

**Comment by wesm** [comment]
Thanks! According to llvm.org, clang-5.0 is still the qualification branch (http://apt.llvm.org/) so whenever 5.0 is promoted to stable we'll upgrade our clang bits to 5.0

**Comment by Licht-T** [comment]
@wesm Now fixed!

**Review by wesm — APPROVED** [review]
+1, thank you! Will merge on green build

**Comment by wesm** [comment]
Couple flake8 warnings:

```
+flake8 --count /home/travis/build/apache/arrow/python/pyarrow
/home/travis/build/apache/arrow/python/pyarrow/tests/test_convert_pandas.py:22:1: F401 'unittest' imported but unused
/home/travis/build/apache/arrow/python/pyarrow/tests/test_convert_pandas.py:1115:16: E231 missing whitespace after ','
```

**Comment by wesm** [comment]
Turns out I can push to your branch, so done

**Comment by Licht-T** [comment]
@wesm Thank you!

**Pull request closed without merging**